### DISCUSSION TITLE
Newbie facing core dumps on fresh run.

### DISCUSSION THREAD

**Discussion by rleed** [romanz/electrs#1054] [Q&A] [answered]
Hi, I've installed deps for, cloned, and built electrs successfully with only one warning. However, I am facing several issues that do not look right, and I am reaching out in the hopes that someone with more experience can put me on the right track. The issues I'm seeing are:
- Ctrl-C not only initiates electrs shutdown, but it also ends in a core dump.
- Electrs initial sync runs until the 180001 - 182000 batch and then core dumps. Consistently, always at the same point.
- Restarting electrs always restarts the initial sync. This seems like it ought not be so.
- Settings that I believe should affect the batch size during the initial sync do not seem to be taking effect, although errors in the settings file are detected and reported so I know the settings file is being read and processed.

I am running on a modern Ubuntu OS with a current Bitcoin Knots backend with txindex=1. I would have installed electrs using apt except despite the extensive documentation about how it should "just work", it wasn't clear to me how to add the PPA. I guess it is so trivial that it wasn't worth mentioning in the docs, but as a newbie in a few areas, I'm drawing a blank on that. Any tips would be welcomed.

**Comment by yancyribbens** [comment]
I've never used Bitcoin Knots with electrs, so I suspect that may be your issue, at least in part.  Also I've never seen a core dump in a Rust application, that's usually encountered in a C or C++ program.  I supposed it's possible if Rust is using FFI to a C/C++ program, although I've never experience that.  Lastly, this seems more of something that ought to be posted as a github issue rather then a discussion.

**Reply by rleed** [reply, marked as answer]
Thanks so much for taking the time to respond thoughtfully! Your answer gave me the clue I needed: No core dump in Rust => must be a dependency issue => recheck the official guide => oops, build-essential was not installed! After fixing that, it is syncing without core dumping, interrupts do not result in a core dump, and it does not restart from "square" one every time. :)

**Reply by Kixunil** [reply]
Now I'd love to know how the hell did you even manage to compile it without `build-essential`. Would be cool if we could detect it sanely to error out early.

Also FYI @yancyribbens electrs uses rocksdb (which is C++) over FFI
